src/app/log-tool/system-setup/clean-data/clean-data.component.ts

```diff
@@ -19,9 +20,13 @@ export class CleanDataComponent implements OnInit {
   dataSubmitted: boolean = false;
   dataExists: boolean = false;
   showEditModal: boolean = false;
+  dataIntervalValid: boolean = false;
   editField: LogToolField;
   individualDataFromCsv: Array<IndividualDataFromCsv>;
   dateExistsForEachCsv: boolean;
+  intervalForSecondsSub: Subscription;
```

Comment:
Can we get rid of the intervalForSeconds Sub and BS? Keep the intervalForSeconds component property and pass it to submitIndividualCSVData (make it a property of that object).

With the current setup, intervalForSeconds is set when the user makes a selection, and also set redundantly when that intervalForSeconds sub is emitted to.